burn-core/src/nn/lstm/lstm.rs

```diff
@@ -0,0 +1,295 @@
+use burn_tensor::activation;
+
+use crate as burn;
+
+use crate::config::Config;
+use crate::module::Module;
+use crate::nn::lstm::gate_controller;
+use crate::nn::Initializer;
+use crate::nn::LinearConfig;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+
+use super::gate_controller::GateController;
+
+#[derive(Config)]
+pub struct LSTMConfig {
+    /// The size of the input features.
+    pub d_input: usize,
+    /// The size of the hidden state.
+    pub d_hidden: usize,
+    /// If a bias should be applied during the Lstm transformation.
+    pub bias: bool,
+    /// Lstm initializer, should probably be Xavier or small random numbers
```

Comment:
Please add a TODO and link this issue: https://github.com/burn-rs/burn/issues/371 so we can come back and update it. Also please update the issue ticket to mention about addition work to update NN lstm.